Framework/Parallel/inc/MantidParallel/IO/EventLoaderHelpers.h

```diff
@@ -0,0 +1,149 @@
+#ifndef MANTID_PARALLEL_EVENTLOADERHELPERS_H_
+#define MANTID_PARALLEL_EVENTLOADERHELPERS_H_
+
+#include "MantidKernel/make_unique.h"
+#include "MantidParallel/Communicator.h"
+#include "MantidParallel/DllConfig.h"
+#include "MantidParallel/IO/Chunker.h"
+#include "MantidParallel/IO/NXEventDataLoader.h"
+
+namespace Mantid {
+namespace Parallel {
+namespace IO {
+
+/** Private parts of EventLoader.
+
+  @author Simon Heybrock
+  @date 2017
+
+  Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+namespace EventLoader {
+/// Read number of events in given banks from file.
+std::vector<size_t> readBankSizes(const H5::Group &group,
+                                  const std::vector<std::string> &bankNames) {
+  std::vector<size_t> bankSizes;
+  for (const auto &bankName : bankNames) {
+    const H5::DataSet dataset = group.openDataSet(bankName + "/event_id");
+    const H5::DataSpace dataSpace = dataset.getSpace();
+    bankSizes.push_back(dataSpace.getSelectNpoints());
+  }
+  return bankSizes;
+}
+
+H5::DataType readDataType(const H5::Group &group,
+                          const std::vector<std::string> &bankNames,
+                          const std::string &name) {
+  return group.openDataSet(bankNames.front() + "/" + name).getDataType();
+}
+
+template <class IndexType, class TimeZeroType, class TimeOffsetType>
+void load(
+    const Chunker &chunker,
+    NXEventDataSource<IndexType, TimeZeroType, TimeOffsetType> &dataSource) {
+  const size_t chunkSize = chunker.chunkSize();
+  const auto &ranges = chunker.makeLoadRanges();
+  // const auto &rankGroups = chunker.makeRankGroups();
+  std::vector<int32_t> event_id(2 * chunkSize);
+  std::vector<TimeOffsetType> event_time_offset(2 * chunkSize);
+
+  // TODO Create parser. Constructor arguments:
```

Comment:
When is TODO to be implemented?

Reply:
https://github.com/mantidproject/mantid/tree/20887_integrate_parallel_event_loader